### ISSUE TITLE
ALLOW_EMPTY_STRING_KEYS doesn't work when skip=True for dpath.util.get

### ISSUE THREAD

**Issue by srids** [akesterson/dpath-python#82]
facing this issue :

```
import dpath
dpath.options.ALLOW_EMPTY_STRING_KEYS=True
tdict = {'': 'fefa1c8e-9c9f-4da4-8964-a7d93274c101'}
dpath.util.get(tdict, '')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.6/site-packages/dpath/util.py", line 103, in get
    for item in search(obj, glob, yielded=True, separator=separator):
  File "/usr/local/lib/python3.6/site-packages/dpath/util.py", line 143, in _search_yielded
    globlist = __safe_path__(glob, separator)
  File "/usr/local/lib/python3.6/site-packages/dpath/util.py", line 25, in __safe_path__
    key = elem[0]
IndexError: string index out of range
```

or with a similar test as

```
import dpath
dpath.options.ALLOW_EMPTY_STRING_KEYS=True
tdict = {
    "Empty": {
        "": {
            "Key": ""
        }
    }
}
dpath.util.get(tdict, 'Empty//Key')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.6/site-packages/dpath/util.py", line 103, in get
    for item in search(obj, glob, yielded=True, separator=separator):
  File "/usr/local/lib/python3.6/site-packages/dpath/util.py", line 143, in _search_yielded
    globlist = __safe_path__(glob, separator)
  File "/usr/local/lib/python3.6/site-packages/dpath/util.py", line 25, in __safe_path__
    key = elem[0]
IndexError: string index out of range
```

**Comment by danielpops** [comment]
See #73

**Comment by srids** [comment]
just a note though, the trace is bit different. util.py vs path.py module. different dpath versions?

**Comment by mroth00** [comment]
Was this fixed? I am experiencing the same issue in dpath 1.4.2 / python 3.7 and cannot tell if there is another argument or something I am missing.

```
dpath.options.ALLOW_EMPTY_STRING_KEYS = True
dpath.util.get({"":"foo","key2":"bar"}, 'key2', separator='..')
IndexError: string index out of range
```

**Comment by srids** [comment]
@mroth00 given the pending PRs to be merged for so long, I ended up forking and fixing it (https://github.com/streamsets/dpath-python/commit/a73be9d6eec788ee1a3d8b16df2163ba111afac7) based on one of the PR I saw. You might wanna get the solution out from our repo at https://github.com/streamsets/dpath-python

**Comment by akesterson** [comment]
This was resolved in response to #73 via pull request #96 

Sorry for the incredibly long delay, somehow we hadnt been getting notified about the backlog.
